package/src/main/java/com/microsoft/azure/cosmos/cassandra/CosmosRetryPolicy.java

```diff
@@ -117,49 +127,37 @@ public RetryDecision onWriteTimeout(
             int requiredAcks,
             int receivedAcks,
             int retryNumber) {
-
         return retryManyTimesOrThrow(retryNumber);
     }
-    
+
     private final static Random random = new Random();
     private final int growingBackOffSaltMillis = 2000;
     private final int fixedBackOffTimeMillis;
     private final int growingBackOffTimeMillis;
     private final int maxRetryCount;
+    private boolean useRetryMillisIfAvailable;
 
     private RetryDecision retryManyTimesOrThrow(int retryNumber) {
-
-        RetryDecision retryDecision;
-
-        if (this.maxRetryCount == -1) {
-            retryDecision = RetryDecision.retry(null);
-        } else {
-            if (retryNumber < this.maxRetryCount) {
-                retryDecision = RetryDecision.retry(null);
-            } else {
-                retryDecision = RetryDecision.rethrow();
-            }
-        }
-
-        return retryDecision;
+        return (this.maxRetryCount == -1 || retryNumber < this.maxRetryCount) ?
+                RetryDecision.retry(null) : RetryDecision.rethrow();
     }
 
-    private RetryDecision retryManyTimesWithBackOffOrThrow(int retryNumber) throws InterruptedException {
+    // Example exceptionString:
+    // "com.datastax.driver.core.exceptions.OverloadedException: Queried host (babatsai.cassandra.cosmos.azure.com/40.65.106.154:10350)
+    // was overloaded: Request rate is large: ActivityID=98f98762-512e-442d-b5ef-36f5d03d788f, RetryAfterMs=10, Additional details='
+    public int getRetryAfterMs(String exceptionString){
+        String[] exceptions = exceptionString.toString().split(",");
+        if (exceptions.length < 2) return -1;
```

Comment:
Remove this condition and split on "=" for all the properties returned in the exception string.
e.g.

string[] tokens = exceptionString.toString().split(",");
foreach (string token in tokens)
{
    string[] kvp = token.split("=");
    if (kvp.length != 2) continue;
    if (kvp[0].trim().equals("RetryAfterMs"))
    {
        String value = kvp[1];
        return Integer.parseInt(value);
    }
}